enhancements/cluster-capabilities-api.md

```diff
@@ -0,0 +1,357 @@
+
+## Summary
+
+As we add more deployment topologies for OpenShift, we need a way to
+communicate expectations to the operators managing the cluster. This
+enhancement describes an API for expressing the expected
+"capabilities" of a cluster, including aspects such as how hard
+operators should work to provide high-availability and whether
+operators should configure their operands in a way to minimize
+resource consumption for especially constrained environments.
+
+## Motivation
+
+We want to avoid having cluster profiles use different manifests to
+configure operators managed by the cluster-version-operator, and
+instead rely on in-cluster APIs. This approach will make the same
+information available to operators managed by either the
+`cluster-version-operator` or `operator-lifecycle-manager`.
+
+We do not want to expose the cluster profile name directly to most of
+the operators, because we do not want to have to update operators each
+time we add a new profile.
+
+We want the settings to be global to the cluster, rather than using
+operator-specific APIs, because that simplifies the integration with
+the installer and allows them to be applied consistently in all
+operators that need them.
+
+### Goals
+
+- Expose capability expectations to operators running in a cluster so
+  they can configure their operands appropriately.
+
+### Non-Goals
+
+- Expose cluster profile names to operators.
+- Provide a mutable API for changing the cluster capabilities (at
+  least for now).
+- Predict other capability settings that we might need to expose.
+
+## Proposal
+
+### Cluster Operators
+
+A new [informational
+resource](https://docs.openshift.com/container-platform/4.6/installing/install_config/customizations.html#informational-resources_customizations),
+`capabilities.config.openshift.io`, is created to hold details about
+the configured features of a cluster. It will include one field to
+start, and may be expanded over time.
+
+The `highAvailabilityMode` field will use an enum to express the
+high-availability expectations. Possible values are `full` and
+`none`. The default is `full`, which represents the behavior operators
+have today in a "normal" cluster. The `none` mode will be used in
+single-node deployments (developer and production), and the operator
+should not configure the operand for highly-available
+operation.
+
+The new cluster-wide `capabilities` API will supersede existing
+unsupported APIs, such as the flags to disable high-availability
+requirements for cluster-etcd-operator and
+cluster-authentication-operator.
+
+If other *supported* APIs for configuring resource use exist, they
+take precedence over this new cluster-wide `capabilities` API. The
+cluster-wide API should be treated as a default for cases where
+operator-specific values are not set, a replacement for unsupported
+APIs, and a way to avoid adding new operator-specific APIs.
+
+### Operator Lifecycle Manager Operators
```

Comment:
@mrunalp @ecordell Does this section match what you had in mind for the OLM? We don't have to work out all of the details, as long as we have a general approach in mind.